src/develop/develop.c

```diff
@@ -1553,11 +1553,16 @@ int dt_dev_distort_backtransform(dt_develop_t *dev, float *points, size_t points
 int dt_dev_distort_transform_plus(dt_develop_t *dev, dt_dev_pixelpipe_t *pipe, int pmin, int pmax,
                                   float *points, size_t points_count)
 {
+  dt_pthread_mutex_lock(&dev->history_mutex);
```

Comment:
This looks like a dirty hack to me, i'm afraid there are more issues caused by same cause as this one.
So i'd sat, deeper investigation is required...